movement/watch_faces/sensor/medical_thermometer_face.c

```diff
@@ -0,0 +1,151 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <limits.h>
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+#include "medical_thermometer_face.h"
+#include "thermistor_driver.h"
+
+static float _medical_thermometer_face_update(bool in_fahrenheit) {
+    thermistor_driver_enable();
+    float temperature_c = thermistor_driver_get_temperature();
+    char buf[14];
+    if (in_fahrenheit) {
+        sprintf(buf, "%4.1f#F", temperature_c * 1.8 + 32.0);
+    } else {
+        sprintf(buf, "%4.1f#C", temperature_c);
+    }
+    watch_display_string(buf, 4);
+    thermistor_driver_disable();
+    return temperature_c;
+}
+
+static void _medical_thermometer_face_clear(int last[]) {
+    for (size_t i = 0; i < LAST_SIZE; i++) {
+        last[i] = INT_MIN;
+    }
+}
+
+void medical_thermometer_face_setup(movement_settings_t *settings, uint8_t watch_face_index, void ** context_ptr) {
+    (void) settings;
+    (void) watch_face_index;
+    if (*context_ptr == NULL) {
+        *context_ptr = malloc(sizeof(medical_thermometer_state_t));
+        memset(*context_ptr, 0, sizeof(medical_thermometer_state_t));
+    }
+}
+
+void medical_thermometer_face_activate(movement_settings_t *settings, void *context) {
+    (void) settings;
+    medical_thermometer_state_t *state = (medical_thermometer_state_t *)context;
+    state->mode = MODE_NORMAL;
+    _medical_thermometer_face_clear(state->last);
+    watch_display_string("TE", 0);
+}
+
+bool medical_thermometer_face_loop(movement_event_t event, movement_settings_t *settings, void *context) {
+    medical_thermometer_state_t *state = (medical_thermometer_state_t *)context;
+
+    switch (event.event_type) {
+        case EVENT_ACTIVATE:
+            _medical_thermometer_face_update(settings->bit.use_imperial_units);
+            break;
+        case EVENT_TICK:
+            if (watch_rtc_get_date_time().unit.second % 5 == 0) {
+                switch (state->mode) {
+                    case MODE_NORMAL:
+                        _medical_thermometer_face_update(settings->bit.use_imperial_units);
+                        break;
+                    case MODE_ALARM:
+                        for (size_t i = LAST_SIZE - 1; i > 0; i--) {
+                            state->last[i] = state->last[i - 1];
+                        }
+                        state->last[0] = roundf(_medical_thermometer_face_update(settings->bit.use_imperial_units) * 10.0f);
```

Comment:
What is the impact of this rounding and multiplication on the accuracy of the measurements?

Reply:
I'm rounding to the first decimal place and store the result in an integer, because you can't really compare floats in C.